lib/main.js

```diff
@@ -47,108 +42,215 @@ module.exports.runSass = function (options, describe, it) {
+}
+
+
+var parseModule = function (rule, ctx) {
+  if (rule.type === 'charset') { return parseModule; }
+  if (rule.type === 'comment') {
+    var text = rule.comment.trim();
+    if (!text) { return parseModule; }
+    if (text.substring(0, 10) === '# Module: ') {
+      finishCurrentModule(ctx);
+      ctx.currentModule = { module: text.substring(10), tests: [] };
+      return parseTest;
+    }
+    parseError('Unexpected comment "' + text + '"', 'module header', rule.position);
+  }
+  parseError('Unexpected rule type "' + rule.type + '"', 'module header', rule.position);
+};
+
+
+var parseTest = function (rule, ctx) {
+  if (rule.type === 'comment') {
+    var text = rule.comment.trim();
+    if (!text) { return parseTest; }
+    if (text.match(/^-+$/)) {
+      return parseTest;
+    }
+    if (text.substring(0, 6) === 'Test: ') {
+      finishCurrentTest(ctx);
+      ctx.currentTest = { test: text.substring(6), assertions: [] };
+      return parseAssertion;
+    }
+    return parseModule(rule, ctx);
+  }
+  parseError('Unexpected rule type "' + rule.type + '"', 'test', rule.position);
 };
 
 
-var parseSelector = module.exports.parseSelector = function (sel) {
-  var rule = selectorParser.parse(sel).rule;
-  var ret = {output: false};
-  var attrName = function (rule, name) {
-    return (rule &&
-            rule.attrs &&
-            rule.attrs.length == 1 &&
-            rule.attrs[0].name === name
-           );
-  };
-  if (attrName(rule, 'data-module')) {
-    ret.module = rule.attrs[0].value;
-    rule = rule.rule;
-    if (attrName(rule, 'data-test')) {
-      ret.test = rule.attrs[0].value;
-      rule = rule.rule;
-      if (attrName(rule, 'data-assert')) {
-        ret.output = true;
-        ret.assert = rule.attrs[0].value;
-        rule = rule.rule;
-        if (rule.classNames.length === 1 &&
-            _.contains(['input', 'expect'], rule.classNames[0])) {
-          ret.type = rule.classNames[0];
-          return ret;
-        }
-      } else if (rule.classNames.length === 1 &&
-                 rule.classNames[0].slice(0, 7) === 'assert-') {
-        ret.assert = rule.classNames[0].slice(7);
-        return ret;
+var parseAssertion = function (rule, ctx) {
+  if (rule.type === 'comment') {
+    var text = rule.comment.trim();
+    if (!text) { return parseAssertion; }
+    if (text.substring(0, 2) === '✔ ') {
+      finishCurrentAssertion(ctx);
+      ctx.currentAssertion = {
+        description: text.substring(2),
+        passed: true
+      };
+      return parseAssertion;
+    } else if (text.substring(0, 10) === '✖ FAILED [') {
+      finishCurrentAssertion(ctx);
+      var endAssertionType = text.indexOf(']');
+      ctx.currentAssertion = {
+        description: text.substring(endAssertionType + 2).trim(),
+        passed: false,
+        assertionType: text.substring(10, endAssertionType),
+      };
+      return parseFailureDetail;
+    } else if (text.substring(0, 8) === 'ASSERT: ') {
```

Comment:
`'ASSERT: '` should probably be stored in a constant at the top of the file. I believe it’s never a good idea to have inlined constants.

```js
if (text.substring(0, ASSERT_TOKEN.length) === ASSERT_TOKEN) {
  // …
}
```